src/jit/block.cpp

```diff
@@ -595,9 +596,14 @@ void BasicBlock::CloneBlockState(Compiler* compiler, BasicBlock* to, const Basic
 
     for (GenTreePtr fromStmt = from->bbTreeList; fromStmt != nullptr; fromStmt = fromStmt->gtNext)
     {
-        compiler->fgInsertStmtAtEnd(to,
-                                    compiler->fgNewStmtFromTree(compiler->gtCloneExpr(fromStmt->gtStmt.gtStmtExpr)));
+        auto newExpr = compiler->gtCloneExpr(fromStmt->gtStmt.gtStmtExpr, 0, varNum, varVal);
+        if (!newExpr)
+        {
+            return false;
+        }
```

Comment:
> It appears that later on that we have a noway assert that this method always returns true.

Yes, the loop cloner currently assumes success here and the noway assert preserves that.  The loop unroller currently [abandons the partial clone if `cloneExpr` returns nullptr](https://github.com/dotnet/coreclr/blob/a307cae611968ff1ce1fdd7c6a084ee6364586fe/src/jit/optimizer.cpp#L3141-L3149), and I plan to [preserve that also](https://github.com/JosephTremoulet/coreclr/blob/a34bd8aa05015536a861e8f8caf234514269725b/src/jit/optimizer.cpp#L3098-L3105) in implementing #7843, where I intend for it to call `gtCloneExpr` through `CloneBlockState` like the loop unroller [currently does](https://github.com/dotnet/coreclr/blob/81a1b87909cd27a7313a124f29c503595f2af117/src/jit/optimizer.cpp#L4651) rather than directly, to share the other logic in `CloneBlockState`.